### DISCUSSION TITLE
How to set a global variant or write a public function in Nextflow that can be used in all module files?

### DISCUSSION THREAD

**Discussion by lishengting** [nextflow-io/nextflow#4529] [Q&A]
I have a variant 'project' and a function get_prj() in a file: 
**prj.nf**
```

project = ""
if (!params.project.equalsIgnoreCase("none")) {
	project = "-P " + params.project
}

def get_prj() {
	if (!params.project.equalsIgnoreCase("none")) {
		return "-P " + params.project
	}else{
		return "";
	}
}

```

I want to use them in another 20 module files. 
How could I do this?

I tried 
`include {get_prj} from "./prj" `

But it doesn't work, always says 

>  Unknown process directive: `get_prj`

or

> No such variable: project

Thanks!

**Comment by mribeirodantas** [comment]
You should use `include` to include **Nextflow processes**, not regular functions. For what you're trying to do, check the following link to learn more about the `lib` folder:https://training.nextflow.io/advanced/structure/#lib.
